packages/docusaurus/src/server/index.ts

```diff
@@ -100,6 +100,7 @@ export async function loadContext(
   return {
     siteDir,
     generatedFilesDir,
+    i18nDir,
```

Comment:
wonder if we can't move this under the `I18n` object directly instead of introducing a new context key?

```ts
export type I18n = DeepRequired<I18nConfig> & {currentLocale: string; currentLocalePath: string};
```

?

Reply:
Because we have `siteDir`, `generatedFilesDir `, and `outDir`, it makes sense to put `i18nDir` here?

More pratically, `I18n` is accessible to the client, but the client shouldn't be aware of the file path, so it's just excess bundle size generated.

Reply:
> Because we have siteDir, generatedFilesDir , and outDir, it makes sense to put i18nDir here?

My idea is that if we can namespace related things together it will be better in the long term? This will be used by plugin authors

>  I think we should even have a new staticDirs context here

yes we'll likely want that 👍 

> More pratically, I18n is accessible to the client, but the client shouldn't be aware of the file path

We could filter it in the emitted client file if needed 

> It's also not part of the user config (unlike i18n.path)

Yes, but we'll likely want to make the per-locale dir configurable too later, as it's not flexible enough to expect localization dirs to be all in the i18n dir. Some users would prefer to have a `_fr` folder at the root of their monorepo for example

Reply:
> My idea is that if we can namespace related things together it will be better in the long term? This will be used by plugin authors

Strong agree. I feel like we have too many things scattered at the root.

> we'll likely want to make the per-locale dir configurable too later,

That's already included in the code as a TODO, and also why `localizationDir` includes the locale as a part. In the future, when we have `i18n.locales.fr.path`, we can simply use that to replace our current approach of using `locale`. Each locale's path will be relative to `i18n.path`.

Reply:
But in the future I'd rather not ask plugin authors to have to use a long thing like `i18n.localeConfigs[i18n.currentLocale].path`, that would be simpler to just use `i18n.localizationDir`?

Also wonder if we shouldn't use plurals today. Does it make sense to have `fr-FR` + `fr` as localized dirs at the same time, and implement a fallback?

Reply:
They would never use `i18n.localeConfigs[i18n.currentLocale].path`, though—`i18n.localizationDir` is already the root dir for the current locale, context-aware.

The only use-case is like for the docs plugin's versioning CLI, where it needs to statically know the file location of every locale, instead of relying on the _current_ locale. I think we can expose a util function for that.